lib/diagnostics_channel.js

```diff
@@ -136,9 +138,144 @@ function hasSubscribers(name) {
   return channel.hasSubscribers;
 }
 
+class StoreBinding {
+  constructor(store, build = (v) => v) {
+    this.store = store;
+    this.build = build;
+    this.stack = [];
+
+    this.onEnter = this.onEnter.bind(this);
+    this.onExit = this.onExit.bind(this);
+  }
+
+  onEnter(data) {
+    this.stack.push(this.store.getStore());
+    this.store.enterWith(this.build(data));
```

Comment:
Should we avoid exceptions from build to escape here?

Reply:
Probably, yes.